components/BaseMinMaxSlider.vue

```diff
@@ -0,0 +1,218 @@
+<script setup lang="ts">
+const props = withDefaults(
+  defineProps<{
+    min: number;
+    max: number;
+    step: number;
+    minValue: number;
+    maxValue: number;
+    label: string;
+    unit?: string;
+  }>(),
+  {
+    step: 1,
+    unit: "",
+  }
+);
+
+const emit = defineEmits<{
+  (e: "update:minValue", sliderMinValue: number): void;
+  (e: "update:maxValue", sliderMaxValue: number): void;
+}>();
+
+const sliderDifference = computed(() => {
+  return Math.abs(props.maxValue - props.minValue);
+});
+
+const differencePercent = computed(() => {
+  return ((sliderDifference.value - props.min) / (props.max - props.min)) * 100;
+});
+
+const leftPercent = computed(() => {
+  return ((props.minValue - props.min) / (props.max - props.min)) * 100;
+});
+
+const rightPercent = computed(() => {
+  return 100 - ((props.maxValue - props.min) / (props.max - props.min)) * 100;
+});
+
+const computedStyle = computed(() => {
+  return {
+    "--width": `${differencePercent.value}%`,
+    "--progressLeft": `${leftPercent.value}%`,
+    "--progressRight": `${rightPercent.value}%`,
+  };
+});
```

Comment:
here I still return the css variables cuz the values are used by the ::before pseudoelement so I could not bind the values directly in the template

Reply:
it's ok and looks better than before 👌 
but what I wanted to suggest is to use `v-bind` directly in the `style` (I don't understand why I can't find any info about it in the official Vue docs 🤷 ):
<img width="346" alt="Zrzut ekranu 2023-07-17 o 09 37 49" src="https://github.com/MartaSolo/nuxt3-playground/assets/72549525/dfe4c08a-dd83-43f8-9500-8d57c70628a3">

Reply:
Thank you :), another thing that I had no idea can be done :), so now the component is much simpler :)